### DISCUSSION TITLE
Modelling a flap using custom degree of freedom approach

### DISCUSSION THREAD

**Discussion by vaibhavraghavan** [capytaine/capytaine#488] [Q&A] [answered]
Hello,

I am trying to model a bottom fixed flap as a way to understand the custom degrees of freedom approach. How can I obtain the vector shape function for the unit rotation at the bottom? I understand that the easiest way to do this is to set the rotation center at the bottom of the flap, but I would like to look at it from a custom degrees of freedom/generalized modes approach.

**Comment by mancellin** [comment, marked as answer]
If I understand correctly, you want to learn how to manual defined degrees of freedom while modelling a rigid flap.

In Capytaine, a degree of freedom is defined as a vector at the center of each face describing how this panel is moving around its mean position.
As an example, the rigid-body rotations can be defined by taking a cross-product in the following way:
```python
body.dofs["my_pitch"] = np.array([
                                  np.cross([x0-x, y0-y, z0-z], [0, 1, 0])
                                  for (x, y, z) in body.mesh.faces_centers
                                 ])
```
where `(x0, y0, z0)` is the center of rotation.

**Reply by vaibhavraghavan** [reply]
Thanks! This is great!
